### PR TITLE
Gender on user profile pull request

### PR DESCRIPTION
**Pull request by AlineTheAlien**
This pull request includes: 

- Implementation of the gender icon on a user profile. This takes the information of the user whose gender we are interested in from the server, and depending on the gender displays the appropriate icon next to the user's name on their user profile. The get and set gender mapper methods are also included. 

- Test cases for the get and set gender methods, that test whether the methods receive the correct response from the server. These tests are included in the UserMapperTests class.

- Refactoring of the user profile. The last user profile wasn't too aesthetically pleasing, therefore it was updated and modified to follow more conventional mobile application interface design. 

- User icon on the bottom toolbar of the menu bar can be tapped to access the user profile of the user who is currently using the app. This allows a user to view their own user profile while using the application.

### CODE REVIEW

File: mRides-app/UserProfileActivity.cs [hunk 1]
```diff
@@ -50,6 +50,26 @@ protected override void OnCreate(Bundle savedInstanceState)
             long userGSD = UserMapper.getInstance().GetGSD(id);
             gsdText.Text = "$ " + userGSD.ToString() + " GSD";
 
+            ImageView genderIcon = FindViewById<ImageView>(Resource.Id.genderImage);
+            //userMapper.setGender(id, "male");
```
Comment (ghost) [inline mRides-app/UserProfileActivity.cs]: You should remove this comment

File: mRides-app/MapActivity.cs [hunk 2]
```diff
@@ -440,6 +437,16 @@ protected override void OnResume()
                 Intent i = new Intent(this, typeof(ChatListActivity));
                 StartActivity(i);
             };
+
+            //MenuBar
+            var userProfileButton = FindViewById<ImageButton>(Resource.Id.menu_user);
+            userProfileButton.Click += delegate
+            {
+                Intent i = new Intent(this, typeof(UserProfileActivity));
+                i.PutExtra("id", User.currentUser.id.ToString()); //not sure if we store this userID
```
Comment (ghost) [inline mRides-app/MapActivity.cs]: Don't forget to remove this one too

File: mRides-app/Mappers/UserMapper.cs [hunk 3]
```diff
@@ -199,6 +199,49 @@ public long setGSD(int id, long gsdAmount)
...
+
+            //Response variable
+            var response = client.Execute(request);
+
+            string genderObtained = response.Content;
+            return genderObtained;
+        }
+
+        /**
+        * Change a user's gender
```
Comment (ghost) [inline mRides-app/Mappers/UserMapper.cs]: Change this to an XML comment for the doc

File: mRides-app/UserProfileActivity.cs (hunk 1 above)
Comment (AlineTheAlien) [reply mRides-app/UserProfileActivity.cs]: Totally missed it, thank you

**Pull request merged**